src/webgpu/api/operation/render_pass/resolve.spec.ts

```diff
@@ -0,0 +1,206 @@
+export const description = `API Operation Tests for RenderPass StoreOp.
+
+Tests a render pass with a resolveTarget resolves correctly for many combinations of:
+  - number of color attachments, some with and some without a resolveTarget
+  - renderPass storeOp set to {‘store’, ‘clear’}
+  - resolveTarget mip level set to {‘0’, base mip > ‘0’}
+  - resolveTarget base array layer set to {‘0’, base layer > '0'} for 2D textures
+  TODO: test all renderable color formats
+`;
+
+import { params, poptions } from '../../../../common/framework/params_builder.js';
+import { makeTestGroup } from '../../../../common/framework/test_group.js';
+import { GPUTest } from '../../../gpu_test.js';
+
+// Test with a zero and non-zero base mip level.
+const kBaseMipLevel: number[] = [0, 1];
+
+// Test with different numbers of color attachments.
+type NumColorAttachments = 2 | 4;
+const kNumColorAttachments: NumColorAttachments[] = [2, 4];
+// Test with a zero and non-zero base array layer.
+const kBaseArrayLayers: number[] = [0, 1];
+
+const kSlotsToResolve: number[][] = [
+  [0, 2],
+  [1, 3],
+  [0, 1, 2, 3],
+];
+
+const kStoreOps: GPUStoreOp[] = ['clear', 'store'];
+
+const kSize = 4;
+const kFormat: GPUTextureFormat = 'rgba8unorm';
+export const g = makeTestGroup(GPUTest);
+
+g.test('render_pass_resolve')
+  .params(
+    params()
+      .combine(poptions('numColorAttachments', kNumColorAttachments))
+      .combine(poptions('slotsToResolve', kSlotsToResolve))
+      .combine(poptions('storeOperation', kStoreOps))
```

Comment:
With inline values, the trick to get the types to infer properly here is:
```suggestion
      .combine(poptions('storeOperation', ['clear', 'store'] as const))
```